Description: Fix unused result compiler warning.
 Fix the following compiler warnings, which appear when -D_FORTIFY_SOURCE=2 is
 passed to gcc:
 .
 mem_freebsd.c: In function ‘mem_init’:
 mem_freebsd.c:38:2: warning: ignoring return value of ‘seteuid’, declared with
 attribute warn_unused_result [-Wunused-result]
  seteuid(getuid());
  ^
 mem_freebsd.c:39:2: warning: ignoring return value of ‘setegid’, declared with
 attribute warn_unused_result [-Wunused-result]
  setegid(getgid());
  ^
Author: Doug Torrance <dtorrance@monmouthcollege.edu>
Last-Update: 2014-10-10

--- a/src/mem_freebsd.c
+++ b/src/mem_freebsd.c
@@ -27,6 +27,8 @@
 /* initialize function */
 void mem_init(void)
 {
+	int s;
+
 	kvm_data = kvm_open(NULL, NULL, NULL, O_RDONLY, "kvm_open");
 
 	if (kvm_data == NULL) {
@@ -35,8 +37,19 @@
 	}
 
 	/* drop setgid & setuid (the latter should not be there really) */
-	seteuid(getuid());
-	setegid(getgid());
+	s = seteuid(getuid());
+	if (s == -1) {
+		fprintf(stderr, "seteuid(getuid()) failed: %s\n",
+			strerror(errno));
+		exit(1);
+	}
+
+	s = setegid(getgid());
+	if (s == -1) {
+		fprintf(stderr, "setegid(getgid()) failed: %s\n",
+			strerror(errno));
+		exit(1);
+	}
 
 	if (geteuid() != getuid() || getegid() != getgid()) {
 		fprintf(stderr, "unable to drop privileges");
